bigchaindb/core.py

```diff
@@ -132,46 +137,59 @@ def write_transaction(self, signed_transaction, durability='soft'):
         response = r.table('backlog').insert(signed_transaction, durability=durability).run(self.conn)
         return response
 
-    def get_transaction(self, txid):
+    def get_transaction(self, txid, include_status=False):
         """Retrieve a transaction with `txid` from bigchain.
 
-        Queries the bigchain for a transaction that was already included in a block.
+        Queries the bigchain for a transaction, if it's in a valid or invalid
+        block.
 
         Args:
             txid (str): transaction id of the transaction to query
+            include_status (bool): also return the status of the transaction
+                                   the return value is then a tuple: (tx, status)
 
         Returns:
             A dict with the transaction details if the transaction was found.
-
-            If no transaction with that `txid` was found it returns `None`
+            Will add the transaction status to payload ('valid', 'undecided',
+            or 'backlog'). If no transaction with that `txid` was found it 
+            returns `None`
         """
 
+        response, tx_status = None, None
+
         validity = self.get_blocks_status_containing_tx(txid)
 
         if validity:
             # Disregard invalid blocks, and return if there are no valid or undecided blocks
             validity = {_id: status for _id, status in validity.items()
                                     if status != Bigchain.BLOCK_INVALID}
-            if not validity:
-                return None
-
-            # If the transaction is in a valid or any undecided block, return it. Does not check
-            # if transactions in undecided blocks are consistent, but selects the valid block before
-            # undecided ones
-            for _id in validity:
-                target_block_id = _id
-                if validity[_id] == Bigchain.BLOCK_VALID:
-                    break
-
-            # Query the transaction in the target block and return
-            response = r.table('bigchain', read_mode=self.read_mode).get(target_block_id)\
-                .get_field('block').get_field('transactions')\
-                .filter(lambda tx: tx['id'] == txid).run(self.conn)
-
-            return response[0]
+            if validity:
```

Comment:
This `if` block can be simplified. If you enter the block the transaction is at least undecided
```python
if validity:
    tx_status = self.TX_UNDECIDED
    for _id in validity:
        if validity[_id] == Bigchain.BLOCK_VALID:
            tx_status = self.TX_VALID
            break
```